helper/resource/testing_example_test.go

```diff
@@ -209,6 +211,67 @@ func ExampleTestCheckResourceAttr_typeString() {
 	resource.TestCheckResourceAttr("example_thing.test", "example_string_attribute", "test-value")
 }
 
+func ExampleTestCheckResourceAttrWith_typeString() {
+	// This function is typically implemented in a TestStep type Check field,
+	// wrapped with ComposeAggregateTestCheckFunc to combine results from
+	// multiple checks.
+	//
+	// Given the following example configuration:
+	//
+	//     resource "example_thing" "test" {
+	//       example_string_attribute = "Very long string..."
+	//     }
+	//
+	// The following TestCheckResourceAttrWith can be written to assert against
+	// the expected state values.
+	//
+	// NOTE: State value checking is only necessary for Computed attributes,
+	//       as the testing framework will automatically return test failures
+	//       for configured attributes that mismatch the saved state, however
+	//       this configuration and test is shown for illustrative purposes.
+
+	// Verify the attribute value string length is above 1000
+	resource.TestCheckResourceAttrWith("example_thing.test", "example_string_attribute", func(value string) error {
+		if len(value) <= 1000 {
+			return fmt.Errorf("example_string_attribute should be longer than 1000 characters")
```

Comment:
Paired with the TestCheckResourceAttrWith error context change above

```suggestion
			return fmt.Errorf("should be longer than 1000 characters")
```